### PR TITLE
Propagate indexing errors to python - and the enduser

### PR DESCRIPTION
**Pull request by ErlendHaa**

### CODE REVIEW

File: python/dlisio/ext/lis.cpp [hunk 1]
```diff
@@ -365,9 +365,10 @@ void init_lis_extension(py::module_ &m) {
     ;
 
     py::class_< lis::record_index >( m, "lis_record_index" )
-        .def( "explicits", &lis::record_index::explicits )
-        .def( "implicits", &lis::record_index::implicits )
-        .def( "size",      &lis::record_index::size )
+        .def( "explicits",    &lis::record_index::explicits )
+        .def( "implicits",    &lis::record_index::implicits )
+        .def( "size",         &lis::record_index::size )
+        .def( "isincomplete", &lis::record_index::is_incomplete)
```
Comment (achaikou) [inline python/dlisio/ext/lis.cpp:370]: why isincomplete and not is_incomplete?

File: lib/include/dlisio/lis/io.hpp [hunk 2]
```diff
@@ -100,11 +101,16 @@ struct record_index {
      */
     bool is_incomplete() const noexcept (true);
 
+    /* Stores the error msg from the (if applicable) failed indexing process
```
Comment (achaikou) [inline lib/include/dlisio/lis/io.hpp:104]: It doesn't store the message, it returns it.

Plus: Propegate -> Prop(a)gate in commit message

File: python/tests/lis/test_physical_layout.py [hunk 3]
```diff
@@ -211,7 +211,7 @@ def test_truncated(filename, assert_error):
     fpath = 'data/lis/layouts/' + filename
     lf = Expected(records=1, ttlr=None, rtlr=None)
     assert_load_correctly(fpath, [lf])
-    assert_error("File likely truncated")
+    assert_error("Indexing failed")
 
 # TODO: all wrong file tests have same error message, may be precised
```
Comment (achaikou) [inline python/tests/lis/test_physical_layout.py:216]: Thanks :D This may be removed now :D

**Review by achaikou — COMMENTED** [review]
Fix what you want to fix and merge

File: python/dlisio/ext/lis.cpp (hunk 1 above)
Comment (ErlendHaa) [reply python/dlisio/ext/lis.cpp:370]: Convension. In c++ it's common to use `is_`. e.g. [1], while in python it's common to to drop the underscore. e.g. [2]
[1] https://en.cppreference.com/w/cpp/types/is_same
[2] https://docs.python.org/3/library/functions.html#isinstance

**Pull request merged**